ambry-api/src/main/java/com.github.ambry/store/StoreKeyFactory.java

```diff
@@ -28,5 +28,13 @@
    * @return The store key created from the stream
    */
   StoreKey getStoreKey(DataInputStream stream) throws IOException;
+
+  /**
+   * The store key created using the input provided, string should be base64 version of
+   * data that would have been read from an input stream
+   * @param base64String The string used to create the store key
+   * @return The store key created from the string
+   */
+  StoreKey getStoreKey(String base64String) throws IOException;
```

Comment:
doesn't need to be base64 (up to the impl) - the docs will need changing too

Reply:
Thought it would be helpful for some guidance on how one should use the interface, but I suppose that's true.